multinode_integration_tests/src/masq_node_cluster.rs

```diff
@@ -91,63 +88,62 @@ impl MASQNodeCluster {
         self.start_mock_node_added_to_cluster(ports, Some(public_key))
     }
 
-    pub fn start_configurable_mock_node_with_public_key(
+    pub fn start_modifiable_mock_node_with_public_key(
         &mut self,
         ports: Vec<u16>,
         public_key: &PublicKey,
-    ) -> MASQMockNodeWithMutableHandle {
-        match self.start_mock_node(
-            ports,
-            MASQMockNodeStarterMode::WithTemporarilyMutableHandle,
-            Some(public_key),
-        ) {
-            Either::Right(mock_node_mutable_handle) => mock_node_mutable_handle,
-            _ => panic!(
-                "This is expected to return MASQMockNodeMutablePrototype but we hit MASQMockNode"
-            ),
-        }
+    ) -> ModifiableMasqMockNode {
+        self.start_mock_node_with_mutable_handle(ports, Some(public_key))
     }
 
     fn start_mock_node_added_to_cluster(
         &mut self,
         ports: Vec<u16>,
         public_key_opt: Option<&PublicKey>,
     ) -> MASQMockNode {
-        match self.start_mock_node(ports, MASQMockNodeStarterMode::Direct, public_key_opt) {
-            Either::Left(masq_mock_node) => {
-                let name = masq_mock_node.name().to_string();
-                self.mock_nodes.insert(name.clone(), masq_mock_node);
-                self.mock_nodes.get(&name).unwrap().clone()
-            }
-            _ => panic!(
-                "This is expected to return MASQMockNode but we hit MASQMockNodeMutablePrototype"
-            ),
-        }
+        let mock_node = self.start_mock_node(ports, public_key_opt);
+        let name = mock_node.name().to_string();
+        self.mock_nodes.insert(name.clone(), mock_node);
+        self.mock_nodes.get(&name).unwrap().clone()
     }
 
     fn start_mock_node(
         &mut self,
         ports: Vec<u16>,
-        mock_node_starter_mode: MASQMockNodeStarterMode,
         public_key_opt: Option<&PublicKey>,
-    ) -> Either<MASQMockNode, MASQMockNodeWithMutableHandle> {
+    ) -> MASQMockNode {
         let index = self.next_index;
         self.next_index += 1;
         MASQMockNodeStarter::start(
             ports,
             index,
             self.host_node_parent_dir.clone(),
-            mock_node_starter_mode,
             public_key_opt,
             self.chain,
         )
     }
 
-    pub fn finalize_mock_node_with_mutable_handle(
+    fn start_mock_node_with_mutable_handle(
+        &mut self,
+        ports: Vec<u16>,
+        public_key_opt: Option<&PublicKey>,
+    ) -> ModifiableMasqMockNode {
+        let index = self.next_index;
+        self.next_index += 1;
+        MASQMockNodeStarter::start_with_mutable_handle(
+            ports,
+            index,
+            self.host_node_parent_dir.clone(),
+            public_key_opt,
+            self.chain,
+        )
+    }
+
+    pub fn finalize_mutable_mock_node_and_add_it_to_cluster(
```

Comment:
If the parameter has a good name, you don't need to mention what is being added in the method name, because the parameter name will take care of that. Also, it should be evident from the call what is being added to (a cluster). Sooo...how about `finalize_and_add`?